src/test/java/co/rsk/federate/config/PowHSMBookkeepingConfigTest.java

```diff
@@ -46,19 +52,50 @@ void setup() {
     }
 
     @Test
-    void getDifficultyTarget_whenCustomConfigAvailable_shouldReturnCustomConfig() {
+    void getDifficultyTarget_whenPowHSMVersionIsLessThanThreeAndCustomConfigIsPresent_shouldReturnCustomConfig() throws Exception {
         BigInteger customDifficultyTarget = BigInteger.valueOf(1L); 
-        when(config.hasPath(DIFFICULTY_TARGET_PATH)).thenReturn(true);
+        when(hsmClient.getVersion()).thenReturn(2);
         when(config.getString(DIFFICULTY_TARGET_PATH)).thenReturn("1");
 
-        assertEquals(customDifficultyTarget, powHsmBookkeepingConfig.getDifficultyTarget());
+        assertEquals(customDifficultyTarget, powHsmBookkeepingConfig.getDifficultyTarget(hsmClient));
+    }
+
+    @Test
+    void getDifficultyTarget_whenPowHSMVersionIsLessThanThreeAndCustomConfigIsNotPresent_shouldThrowException() throws Exception {
+        when(hsmClient.getVersion()).thenReturn(2);
+        when(config.getString(DIFFICULTY_TARGET_PATH)).thenReturn(null);
+
+        assertThrows(IllegalStateException.class,
+            () -> powHsmBookkeepingConfig.getDifficultyTarget(hsmClient));
+    }
+
+    @Test
+    void getDifficultyTarget_whenPowHSMVersionIsGreaterThanTwo_shouldRetriveConfigValueFromPowHSM() throws Exception {
+        BigInteger difficultyTarget = BigInteger.valueOf(1L); 
```

Comment:
nit: `expectedDifficultyTarget`? This way, when we read

`assertEquals(expectedDifficultyTarget, powHsmBookkeepingConfig.getDifficultyTarget(hsmClient));`

it would be easier.

Reply:
Sounds good, updated!